[Site Isolation] Exposed content rect of subframe view is wrong

szewai opened the pull request:
#### b9473ec1d60adf0b5325f08907893928d2df946f
<pre>
[Site Isolation] Exposed content rect of subframe view is wrong
<a href="https://bugs.webkit.org/show_bug.cgi?id=268647">https://bugs.webkit.org/show_bug.cgi?id=268647</a>
<a href="https://rdar.apple.com/122199122">rdar://122199122</a>

Reviewed by Simon Fraser and Alex Christensen.

On iOS, exposedContentRect of FrameView is used to decide visible area that needs compositing update (see
RenderLayerCompositor::visibleRectForLayerFlushing and RenderLayerCompositor::flushPendingLayerChanges). Currently
exposedContentRect is set via DrawingArea (and each DrawingArea has one content rect), and it is only set on the main
frame view (as child frames can inheirt the value from their parent, see ScrollView::exposedContentRect()). That means
frames in the subframe process do not have the correct content rect and thus do not render. To fix that, now we set the
exposedContentRect of those frames to be the frame rect. This is a temporary fix that enables us to run tests; we are
going to make a complete fix that ensures UI process sends geometry updates to web process in <a href="https://rdar.apple.com/122429810">rdar://122429810</a>.

* LayoutTests/platform/ios/TestExpectations:
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(WebKit::WebPage::updateFrameSize):

Canonical link: <a href="https://commits.webkit.org/274302@main">https://commits.webkit.org/274302@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/64d2a7a7fe879325b1a1e357693b93455987c4e1

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/38646 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/17575 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/40991 "Built successfully") | [⏳ 🛠 wpe ](https://ews-build.webkit.org/#/builders/WPE-Build-EWS "Waiting in queue, processing has not started yet") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/32/builds/34303 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/20380 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/14922 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/32472 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/39219 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/14828 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/33606 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/12868 "Passed tests") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/12852 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/10/builds/34478 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/42451 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/35124 "Passed tests") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/34862 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/38689 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/44/builds/13460 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/11157 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/36894 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/46/builds/15072 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/43/builds/13918 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/5035 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/14549 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

achristensen07 commented on Source/WebKit/WebProcess/WebPage/WebPage.cpp:
```diff
@@ -3210,6 +3210,8 @@ void WebPage::updateFrameSize(WebCore::FrameIdentifier frameID, WebCore::IntSize
 
     frameView->resize(newSize);
+    // FIXME: This should be removed after rdar://122429810 is fixed.
+    frameView->setExposedContentRect(frameView->frameRect());
```
Could we also add ASSERT(m_page->settings().siteIsolationEnabled()) to make it more clear that this won't affect anything when site isolation is off?

szewai committed:
[Site Isolation] Exposed content rect of subframe view is wrong
https://bugs.webkit.org/show_bug.cgi?id=268647
rdar://122199122

Reviewed by Simon Fraser and Alex Christensen.

On iOS, exposedContentRect of FrameView is used to decide visible area that needs compositing update (see
RenderLayerCompositor::visibleRectForLayerFlushing and RenderLayerCompositor::flushPendingLayerChanges). Currently
exposedContentRect is set via DrawingArea (and each DrawingArea has one content rect), and it is only set on the main
frame view (as child frames can inheirt the value from their parent, see ScrollView::exposedContentRect()). That means
frames in the subframe process do not have the correct content rect and thus do not render. To fix that, now we set the
exposedContentRect of those frames to be the frame rect. This is a temporary fix that enables us to run tests; we are
going to make a complete fix that ensures UI process sends geometry updates to web process in rdar://122429810.

* LayoutTests/platform/ios/TestExpectations:
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(WebKit::WebPage::updateFrameSize):

Canonical link: https://commits.webkit.org/274302@main
